sim/src/kernels/ocean_noise.opencl

```diff
@@ -33,14 +33,16 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   *  Generates two uniform random numbers based on the ANSIC Linear Congruential 
   *  Generator.
   */
-float2 ansic_lcg(float* seed_ptr) {
-    long seed = (*seed_ptr);
+float2 ansic_lcg(ulong* seed_ptr) {
+    ulong seed = (*seed_ptr);
+    double denum = 2147483648.0;
```

Comment:
On Intel hw (and possibly other hw) the use of double requires enabling the OpenCL extension cl_khr_fp64:

#pragma OPENCL EXTENSION cl_khr_fp64 : enable